src/jit/function-builder.cc

```diff
@@ -199,7 +200,31 @@ void FunctionBuilder::SetUpLocals(TR::IlBuilder* b, const uint8_t** pc, VirtualS
   // Add placeholders onto the virtual stack. These values should never be actually read, but will
   // eventually be dropped using drop or drop_keep in the function epilogue.
   for (size_t i = 0; i < fn_->param_and_local_types.size(); i++) {
-    stack->Push(nullptr);
+    auto t = fn_->param_and_local_types.at(i);
+    auto locals_offset = fn_->param_and_local_types.size() - fn_->local_count;
+
+    if (i < locals_offset) {
+      auto param_addr = PickPhys(b, locals_offset - i);
+      auto param = b->LoadIndirect("Value", TypeFieldName(t), param_addr);
+      stack->Push(param);
+    } else {
+      switch (t) {
+          case Type::I32:
+            stack->Push(b->ConstInt32(0));
+            break;
+          case Type::I64:
+            stack->Push(b->ConstInt64(0));
+            break;
+          case Type::F32:
+            stack->Push(b->ConstFloat(0.0));
+            break;
+          case Type::F64:
+            stack->Push(b->ConstDouble(0.0));
+            break;
+          default:
+            TR_ASSERT_FATAL(false, "Compiling function with local of unsuported type %d\n", static_cast<int32_t>(t));
```

Comment:
This really shouldn't be an assert, since we want to gracefully fail back to the interpreter if we encounter a type we don't support (i.e. `Type::V128`).

Reply:
You're right. Fixed.